dev/spark_integration/Dockerfile

```diff
@@ -0,0 +1,42 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+FROM maven:3.5.2-jdk-8-slim
+ADD . /apache-arrow
```

Comment:
Add `apache-arrow` as late as a possible. From the point on where you have added it, docker's caching will not be effective anymore. For example, once you move the `apt-get` line before the add, a subsequent build of the image will not run `apt-get` again but use the cached layer on top of the maven image that already is at the state after the `apt-get` call.

Reply:
Great tip, thanks @xhochy !  apt-get takes a long time and I didn't know why it only sometimes ended up being cached.